packages/core/src/index.ts

```diff
@@ -122,73 +128,84 @@ export default class Engine<Name extends string = string> {
 				this.publicParsedRules[name] = rule as RuleNode<Name>
 			}
 		}
+
+		this.publicSituation = {} as Situation<Name>
 	}
 
 	resetCache() {
 		this.cache = emptyCache()
 	}
 
 	setSituation(
-		situation: Partial<Record<Name, PublicodesExpression | ASTNode>> = {},
-		options: { keepPreviousSituation?: boolean } = {},
+		situation: Situation<Name> = {},
+		options: {
+			keepPreviousSituation?: boolean
+			safeMode?: boolean
+		} = {},
 	) {
 		this.resetCache()
 
 		const keepPreviousSituation = options.keepPreviousSituation ?? false
-
-		Object.keys(situation).forEach((name) => {
-			if (!(name in this.baseContext.parsedRules)) {
-				throw new PublicodesError(
-					'EvaluationError',
-					`Erreur lors de la mise à jour de la situation : ${name} n'existe pas dans la base de règle.`,
-					{ dottedName: name },
-				)
-			}
-			if (this.baseContext.parsedRules[name].private) {
-				throw new PublicodesError(
-					'EvaluationError',
-					`Erreur lors de la mise à jour de la situation : ${name} est une règle privée (il n'est pas possible de modifier une règle privée).`,
-					{ dottedName: name },
-				)
-			}
-		})
-
-		// The situation is implemented as a special sub namespace `$SITUATION`,
-		// present on each non-private rules
-		const situationToParse = Object.fromEntries(
-			Object.entries(situation).map(([nom, value]) => [
-				`[privé] ${nom} . $SITUATION`,
-				value && typeof value === 'object' && 'nodeKind' in value ?
-					{ valeur: value }
-				:	value,
-			]),
+		const safeMode = options.safeMode ?? this.baseContext.safeMode ?? false
+		const previousContext = this.context
+
+		this.context =
+			keepPreviousSituation ? this.context : copyContext(this.baseContext)
```

Comment:
question: quel est la différence entre le `baseContext` et `context` ici ?

suggestion: l'intention est plus clair je trouve en explicitant la mutation conditionnée ici : 
```suggestion
        if (!keepPreviousSituation) {
          this.context = copyContext(this.baseContext)
        }
```

Reply:
`baseContext` contient les règles de base, sans la situation. `context` contient celle avec la situation précédente.

Reply:
Pourquoi ne pas juste copier la situation dans ce cas là ? Cela éviterait de copier toutes les `parsedRules`.

Reply:
En soit, ce serait même juste assigné l'objet vide à la situation dans ce cas là non ?